beetle-core/src/main/java/com/xing/beetle/util/RetryExecutor.java

```diff
@@ -106,7 +69,9 @@ public void run() {
     CompletionStage<T> schedule(Throwable error) {
       long delayInMillis = backoff.delayInMillis(attempt++, error);
       if (delayInMillis > 0) {
-        scheduler.delayed(delayInMillis, TimeUnit.MILLISECONDS, executor).execute(this);
+        ScheduledExecutorService ses = Executors.newScheduledThreadPool(1);
```

Comment:
I have a bad feeling about every `Retrying` task creating and destroying its delay executor on each schedule.

I do see that it is a straightforward way of ensuring that each task is rescheduled without possibly colliding with other retries.

Could the scheduler not be a member variable of the parent class?

This might use more concurrency to avoid blocking separate rescheduling requests, but might make things a little more complicated in terms of guaranteeing rescheduling after the specified delay if the shared executor is congested.

Reply:
I like the simplificaton of the RetryExecutor (I had the feeling before that it was over-generalized before in ways that are not required by our use case).

Reply:
Yeah, let's make it global for all retries and maybe give it more than 1 thread.

Reply:
LGTM